go.mod: upgrade twpayne/go-geom

otan opened the pull request:
This fixes ARM specific bugs.

Release note (bug fix): Intersection spatial operations would previously
potentially return incorrect results on ARM. This is now resolved.

Release justification: bug fix for existing functionality

Resolves #72226

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/86126)
<!-- Reviewable:end -->

otan committed:
go.mod: upgrade twpayne/go-geom

This fixes ARM specific bugs.

Release note (bug fix): Intersection spatial operations would previously
potentially return incorrect results on ARM. This is now resolved.

Release justification: bug fix for existing functionality

rafiss reviewed (approved):
lgtm! gonna try to bors it since it looks like failures are flakes

bors r+
